clippy_lints/src/feature_name.rs

```diff
@@ -0,0 +1,156 @@
+    fn check_crate(&mut self, cx: &LateContext<'_>, _: &Crate<'_>) {
+        let mut sorted_prefixes = PREFIXES;
+        sorted_prefixes.sort_unstable();
+        assert_eq!(PREFIXES, sorted_prefixes);
+        let mut sorted_postfixes = POSTFIXES;
+        sorted_postfixes.sort_unstable();
+        assert_eq!(POSTFIXES, sorted_postfixes);
+
+        if is_lint_allowed(cx, REDUNDANT_FEATURE_NAMES, CRATE_HIR_ID)
+            && is_lint_allowed(cx, NEGATIVE_FEATURE_NAMES, CRATE_HIR_ID)
+        {
+            return;
+        }
+
+        let metadata = unwrap_cargo_metadata!(cx, REDUNDANT_FEATURE_NAMES, false);
+
+        for package in metadata.packages {
+            let mut features: Vec<&String> = package.features.keys().collect();
+            features.sort();
+            for feature in features {
+                let i = PREFIXES.partition_point(|prefix| prefix < &feature.as_str());
+                if i > 0 && feature.starts_with(PREFIXES[i - 1]) {
+                    if is_negative_prefix(PREFIXES[i - 1]) {
+                        span_lint_and_help(
+                            cx,
+                            NEGATIVE_FEATURE_NAMES,
+                            DUMMY_SP,
+                            &format!(
+                                "the \"{}\" prefix in the feature name \"{}\" is negative",
+                                PREFIXES[i - 1],
+                                feature
+                            ),
+                            None,
+                            &format!(
+                                "consider renaming the feature to \"{}\", but make sure the feature adds functionality",
+                                feature.strip_prefix(PREFIXES[i - 1]).unwrap()
+                            ),
+                        );
+                    } else {
+                        span_lint_and_help(
+                            cx,
+                            REDUNDANT_FEATURE_NAMES,
+                            DUMMY_SP,
+                            &format!(
+                                "the \"{}\" prefix in the feature name \"{}\" is redundant",
+                                PREFIXES[i - 1],
+                                feature
+                            ),
+                            None,
+                            &format!(
+                                "consider renaming the feature to \"{}\"",
+                                feature.strip_prefix(PREFIXES[i - 1]).unwrap()
+                            ),
+                        );
+                    }
+                }
+                for postfix in &POSTFIXES {
+                    if feature.ends_with(postfix) {
+                        span_lint_and_help(
+                            cx,
+                            REDUNDANT_FEATURE_NAMES,
+                            DUMMY_SP,
+                            &format!(
+                                "the \"{}\" postfix in the feature name \"{}\" is redundant",
```

Comment:
"suffix" is really the more common term
```suggestion
                                "the \"{}\" suffix in the feature name \"{}\" is redundant",
```